Taskfile.yml

```diff
@@ -100,22 +102,23 @@ tasks:
       INPUT: '{ "source": { "owner": "foo", "repo": "bar", "access_token": "123"} }'
 
   docker:push:
-    desc: Push the Docker image
+    desc: Push the Docker image.
     cmds:
       - docker push {{.DOCKER_FULL_NAME}}
-      - |-
-        if [ -n "$DOCKER_LATEST" ]; then
-            docker tag {{.DOCKER_FULL_NAME}} {{.DOCKER_BASE_NAME}}:latest
-            # FIXME safety while adding GH actions: disable pushing "latest"
-            exit 1
-            docker push {{.DOCKER_BASE_NAME}}:latest
-            echo Pushed a new \"latest\" tag!
-        else
-            echo Did not push a new \"latest\" tag.
-        fi
-    vars:
-      DOCKER_LATEST:
-        sh: ci/git-latest.sh
+    preconditions:
+      - sh: test -z "$IS_RELEASE" || test -n "$CI"
+        msg: Release tag detected ({{.DOCKER_TAG}}); releases are made only on CI.
+
+  docker:maybe-push-release:
+    desc: If a release tag has been detected, Docker push with the 'latest' tag.
+    cmds:
+      - docker tag {{.DOCKER_FULL_NAME}} {{.DOCKER_BASE_NAME}}:latest
+      - docker push {{.DOCKER_BASE_NAME}}:latest
+    preconditions:
+      - sh: test -n "$CI"
+        msg: This target must run only on CI, not locally.
+    status:
+      - test -z "{{.IS_RELEASE}}"
```

Comment:
For my own education, does the `status` means it will consider the task done when it's not a release (IS_RELEASE is an empty string) and so do nothing, while using a precondition would actually make the task fail?

Reply:
Correct. Since the logic to detect if we are doing a release or not comes from Task and not from the workflow, I need this target to be idempotent and safe to trigger at any time.